src/v2/cookbook/vue-route-change-handling.md

```diff
@@ -0,0 +1,376 @@
+---
+
+## Simple Example
+
+While using [vue-router](https://router.vuejs.org/en/) we often need to refresh the data when a route or query changes while the same component is being reused in the page, to do so we can use a `:key` to force Vue to always recreate the components:
+
+```js
+Vue.component('OrderIndex', {
+  created() {
+    this.loadData();
+  },
+  methods: {
+    loadData() { /* logic to load data */ },
+  },
+});
+```
+
+```html
+<router-view :key="$route.fullPath"></router-view>
+```
+
+This will assure that when going from `/orders` to `/orders?page=2` or from `/orders?page=2` to `/orders?page=1`, a fresh instance of `OrderIndex` component is created, triggering the `created` hook and downloading fresh data. There's also a subtle alternative to it:
+
+```html
+<router-view :key="$route.path"></router-view>
+```
+
+The main difference is that changes on the query eg: going from `/orders` to `/orders?page=2` won't make up a different `key` and therefore, no data is fetched. However going to a different route will keep the same behaviour. 
+
+Sometimes we don't want to recreate the view component so we go with key-less router-view:
+
+```html
+<router-view></router-view>
+```
+
+and `$watch` the `$route` object:
+
+```js
+Vue.component('OrderIndex', {
+  watch: {
+    $route: {
+      immediate: true,
+      handler: 'loadData',
+    },
+  },
+  methods: {
+    loadData() { /* logic to load data */ },
+  },
+});
+```
+
+<p class="tip">Note! [VueRouter](https://router.vuejs.org/en/) supports [named views](https://router.vuejs.org/en/essentials/named-views.html) and [nested routes](https://router.vuejs.org/en/essentials/nested-routes.html) so the component behaviour depends on the assigned `key` to each `router-view`. It's best to avoid mixing different `key` expressions.</p>
+
+Now each time the route changes, `OrderIndex` reloads its data.
+
+<p class="tip">Remember! Don't modify `$route.query` directly - it does not change the url, use `$router.push` or `$router.replace` instead.</p>
+
+## Simple, but not good enough
+While both `:key` methods work, there are some drawbacks to each one. Lets start with:
+
+```html
+<router-view :key="$route.fullPath"></router-view>
+```
+
+The `$route.fullPath` returns the full url including the query e.g. `/orders?page=2&filter=search`, so what happens when you visit from `/orders` to `/orders?page=1`? The page is recreated with new instances of components and the data is reloaded, yet the route is technically the same as before. At this point the only possible workaround would be to cache the results using something like [VueX](https://vuex.vuejs.org) or plain object. We could use `$route.path` as a `key`, but this will require introducing a watcher and handling query changes which sometimes can be difficult and unexpected side effects if implemented incorrectly. There are more examples why `:key` on the `router-view` could introduce more reloads than necessary which we'll cover further.
```

Comment:
_visit -> go_ or _visit orders?page=1 from /orders_ (with the markup you wrote ofc)
No upper X in Vuex
or **a** plain object
if not implemented correctly reads better
I'd remove the last sentence as it doesn't read quite nice and doesn't really say anything